db/migrate/20170503212442_create_itms_zonfp.rb

```diff
@@ -0,0 +1,179 @@
+class CreateItmsZonfp < ActiveRecord::Migration[5.0]
+
+  ZONFP_STATES = %w(prijate schvalene zamietnute)
+
+  def up
+    ZONFP_STATES.each do |zonfp_state|
+      create_table "itms.zonfp_#{zonfp_state}" do |t|
+        t.string :akronym
+        t.string :kod
+        t.string :nazov
+        t.string :zameranie_projektu
+        t.text :popis_projektu
+        t.decimal :percento_ziadane_spolufinancovania
+        t.decimal :suma_ziadana_celkova
+        t.decimal :suma_ziadana_celkova_projektov_generujucich_prijem
+        t.decimal :suma_ziadana_nfp
+        t.decimal :suma_ziadana_vlastnych_zdrojov
+        t.datetime :updated_at
+        t.datetime :created_at
+        t.datetime :datum_predlozenia
+        t.datetime :datum_ziadany_konca_hlavnych_aktivit
+        t.datetime :datum_ziadany_konca_realizacie
+        t.datetime :datum_ziadany_zaciatku_hlavnych_aktivit
+        t.datetime :datum_ziadany_zaciatku_realizacie
+      end
+      change_column "itms.zonfp_#{zonfp_state}", :id , :bigint
+
+      create_table "itms.zonfp_#{zonfp_state}_aktivity_projekty" do |t|
+        t.integer "zonfp_#{zonfp_state}_id", null: false, limit: 8
+        t.string :kod
+        t.string :nazov
+        t.datetime :datum_konca_planovany
+        t.datetime :datum_konca_skutocny
+        t.datetime :datum_zaciatku_planovany
+        t.datetime :datum_zaciatku_skutocny
+      end
+      change_column "itms.zonfp_#{zonfp_state}_aktivity_projekty", :id , :bigint
+
+      create_table "itms.zonfp_#{zonfp_state}_aktivity_projekt_subjekty", primary_key: ['id', "zonfp_#{zonfp_state}_aktivity_projekty_id"] do |t|
+        t.integer :id, null: false, limit: 8
+        t.integer "zonfp_#{zonfp_state}_aktivity_projekty_id", null: false, limit: 8
+        t.string :ico
+        t.string :ine_identifikacne_cislo
+      end
+
+      create_table "itms.zonfp_#{zonfp_state}_formy_financovania", primary_key: ['id', "zonfp_#{zonfp_state}_id"] do |t|
+        t.integer :id, null: false, limit: 8
+        t.integer "zonfp_#{zonfp_state}_id", null: false, limit: 8
+        t.string :kod_zdroj
+        t.string :nazov
+      end
+
+      create_table "itms.zonfp_#{zonfp_state}_formy_financovania_ciele", primary_key: ['id', "zonfp_#{zonfp_state}_formy_financovania_id"] do |t|
+        t.integer :id, null: false, limit: 8
+        t.integer "zonfp_#{zonfp_state}_formy_financovania_id", null: false, limit: 8
+      end
+
+      create_table "itms.zonfp_#{zonfp_state}_hospodarske_cinnosti", primary_key: ['id', "zonfp_#{zonfp_state}_id"] do |t|
+        t.integer :id, null: false, limit: 8
+        t.integer "zonfp_#{zonfp_state}_id", null: false, limit: 8
```

Comment:
pre vsetky cudzie kluce by som ocakaval aj ozajstne cudzie kluce `add_foregn_key` - v dokumentacii to potom generuje prepojenia tabuliek automaticky.